### PR TITLE
py-biom-format: add version 2.1.7 and fix dependencies

### PR DESCRIPTION
**Pull request by erimar77**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 1]
```diff
@@ -14,10 +14,12 @@ class PyBiomFormat(PythonPackage):
     homepage = "https://pypi.python.org/pypi/biom-format/2.1.6"
     url      = "https://pypi.io/packages/source/b/biom-format/biom-format-2.1.6.tar.gz"
 
+    version('2.1.7', sha256='b47e54282ef13cddffdb00aea9183a87175a2372c91a915259086a3f444c42f4')
     version('2.1.6', sha256='8eefc275a85cc937f6d6f408d91b7b45eae854cd5d1cbda411a3af51f5b49b0d')
 
     variant('h5py', default=True, description='For use with BIOM 2.0+ files')
 
+    depends_on('python@:3.6', when='@:2.1.7')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]: According to https://pypi.org/project/biom-format/, biom-format supports both Python 2.6 and 3.7. I think this should be:
```suggestion
    depends_on('python@2.7:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 2]
```diff
@@ -27,4 +29,4 @@ class PyBiomFormat(PythonPackage):
     depends_on('py-scipy@0.13.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]: Numpy is now `@1.9.2:`

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 3]
```diff
@@ -27,4 +29,4 @@ class PyBiomFormat(PythonPackage):
...
     depends_on('py-pandas@0.19.2:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]:
```suggestion
    depends_on('py-pandas@0.20.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 4]
```diff
@@ -27,4 +29,4 @@ class PyBiomFormat(PythonPackage):
...
     depends_on('py-six@1.10.0:', type=('build', 'run'))
-    depends_on('py-pyqi', type=('build', 'run'))
+    depends_on('py-pyqi', type=('build', 'run'), when='^python@2.7.0:2.7.999')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]:
```suggestion
    depends_on('py-pyqi', type=('build', 'run'), when='^python@:2')
```

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 5]
```diff
@@ -14,10 +14,12 @@ class PyBiomFormat(PythonPackage):
...
 
+    version('2.1.7', sha256='b47e54282ef13cddffdb00aea9183a87175a2372c91a915259086a3f444c42f4')
     version('2.1.6', sha256='8eefc275a85cc937f6d6f408d91b7b45eae854cd5d1cbda411a3af51f5b49b0d')
 
     variant('h5py', default=True, description='For use with BIOM 2.0+ files')
 
+    depends_on('python@:3.6', when='@:2.1.7')
     depends_on('py-setuptools', type=('build', 'run'))
     depends_on('py-cython', type='build')
     depends_on('py-h5py', type=('build', 'run'), when='+h5py')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]: It looks like this dependency is actually optional. You could add a variant that defaults to False if you want to decrease the number of dependencies. Also, it should be `@2.2.0:`

File: var/spack/repos/builtin/packages/py-biom-format/package.py (hunk 3 above)
Comment (erimar77) [reply var/spack/repos/builtin/packages/py-biom-format/package.py]: Version 2.1.6 requires pandas >= 0.19.2, should I always default to the latest version if possible?

File: var/spack/repos/builtin/packages/py-biom-format/package.py (hunk 2 above)
Comment (erimar77) [reply var/spack/repos/builtin/packages/py-biom-format/package.py]: Version 2.1.6 requires numpy >= 1.3.0, should I always default to the latest version if possible?

**Comment by adamjstewart** [comment]
When I have 2 versions of a package with different dependency version requirements, I usually take the stricter of the two. If you want to be really correct, you would investigate which version of the package changed the dependency version and have multiple `depends_on` calls, but you can also just choose the stricter restriction if it works for both.

**Comment by adamjstewart** [comment]
Ping @erimar77, see my answer to your question above.

**Comment by alalazo** [comment]
Closing this as stale. @erimar77 Feel free to reopen if you want to continue working on this.

**Pull request closed without merging**